tests/test_cache_blocking.py

```diff
@@ -30,3 +30,25 @@ def test_cache_blocking_remainder(self):
         op_block = SimpleOperator(input_grid, output_grid_block, [eq], cache_blocking=True)
         op_block.apply()
         assert(np.equal(output_grid_block.data, output_grid_noblock.data).all())
+
+    def test_cache_blocking_full_range(self):
+        input_grid = DenseData(name="input_grid", shape=(302, 302), dtype=np.float64)
+        input_grid.data[:] = np.arange(91204, dtype=np.float64).reshape((302, 302))
+        x, t = symbols("x t")
+
+        errors = []
+        for spc_border in range(0, 9):
+            for block_size in range(1, 11):
```

Comment:
We should really parametrize the test itself using `pytest.mark.parametrize`, rather than looping inside the test. The advantage is that pytest will immediately tell you which combinations fail and which ones pass. We can also parametrize the "block-inner-loop" feature to keep the test setup clean.